cdap-app-fabric/src/main/java/co/cask/cdap/common/twill/TwillAppLifecycleEventHandler.java

```diff
@@ -0,0 +1,241 @@
+    this.abortIfNotFull = Boolean.parseBoolean(context.getSpecification().getConfigs().get("abortIfNotFull"));
+    this.runId = context.getSpecification().getConfigs().get("runId");
+    this.programRunId = getProgramRunId(context.getApplicationName());
+
+    // Fetch cConf and hConf from resources jar
+    File cConfFile = new File("resources.jar/resources/" + CDAP_CONF_FILE_NAME);
+    File hConfFile = new File("resources.jar/resources/" + HADOOP_CONF_FILE_NAME);
+
+    if (cConfFile.exists() && hConfFile.exists()) {
+      CConfiguration cConf = CConfiguration.create();
+      cConf.clear();
+
+      Configuration hConf = new Configuration();
+      hConf.clear();
+
+      try {
+        cConf.addResource(cConfFile.toURI().toURL());
+        hConf.addResource(hConfFile.toURI().toURL());
+
+        // Create the injector to inject a program state writer
+        Injector injector = Guice.createInjector(
+          new ConfigModule(cConf, hConf),
+          new ZKClientModule(),
+          new KafkaClientModule(),
+          new DiscoveryRuntimeModule().getDistributedModules(),
+          new AbstractModule() {
+            @Override
+            protected void configure() {
+              bind(RuntimeStore.class).to(RemoteRuntimeStore.class);
+              bind(ProgramStateWriter.class).to(DirectStoreProgramStateWriter.class);
+            }
+          }
+        );
+
+        zkClientService = injector.getInstance(ZKClientService.class);
+        startServices();
+        this.programStateWriter = injector.getInstance(ProgramStateWriter.class);
+
+      } catch (Throwable throwable) {
+        Throwables.propagate(throwable);
+      }
+    } else {
+      LOG.warn("{} and {} were not found in the resources.jar. Not recording program states", CDAP_CONF_FILE_NAME,
+               HADOOP_CONF_FILE_NAME);
+      this.programStateWriter = new NoOpProgramStateWriter();
+    }
+  }
+
+  @Override
+  public TimeoutAction launchTimeout(Iterable<TimeoutEvent> timeoutEvents) {
+    long now = System.currentTimeMillis();
+    for (TimeoutEvent event : timeoutEvents) {
+      LOG.info("Requested {} containers for runnable {}, only got {} after {} ms.",
+               event.getExpectedInstances(), event.getRunnableName(),
+               event.getActualInstances(), System.currentTimeMillis() - event.getRequestTime());
+
+      boolean pass = abortIfNotFull ? event.getActualInstances() == event.getExpectedInstances()
+                                    : event.getActualInstances() != 0;
+      if (!pass && (now - event.getRequestTime()) > abortTime) {
+        LOG.info("No containers for {}. Abort the application.", event.getRunnableName());
+        return TimeoutAction.abort();
+      }
+    }
+    // Check again in half of abort time.
+    return TimeoutAction.recheck(abortTime / 2, TimeUnit.MILLISECONDS);
+  }
+
+ @Override
+  public void started() {
+    super.started();
```

Comment:
no need of calling super, which is no-op, the same for all the following methods